src/e2e-test/java/io/cdap/cdap/ui/utils/Commands.java

```diff
@@ -62,4 +69,107 @@ public static void fitPipelineToScreen() {
     );
   };
 
+  public static void testConnection(String connectionType, String connectionId,
+                                       String projectId, String serviceAccountPath) {
+    fillConnectionCreateForm(connectionType, connectionId, projectId, serviceAccountPath);
+    ElementHelper.clickOnElement(
+            Helper.locateElementByCssSelector(
+                    Helper.getCssSelectorByDataTestId("connection-test-button")), 5);
+
+  }
+
+  public static void createConnection(String connectionType, String connectionId) {
+    fillConnectionCreateForm(connectionType, connectionId, null, null);
+    ElementHelper.clickOnElement(
+            Helper.locateElementByCssSelector(
+                    Helper.getCssSelectorByDataTestId("connection-submit-button")), 5);
+    WaitHelper.waitForPageToLoad();
+
+  }
+
+  public static void deleteConnection(String connectionType, String connectionId) {
+    ElementHelper.clickOnElement(
+            Helper.locateElementByCssSelector(
+                    Helper.getCssSelectorByDataTestId("categorized-connection-type-" + connectionType)));
+
+    String connectionCSSLocator = "connection-" + connectionType + "-" + connectionId;
+    WebElement connectionItem = Helper.locateElementByCssSelector(
+            Helper.getCssSelectorByDataTestId(connectionCSSLocator));
+
+    WebElement connectionItemParentParent = connectionItem.findElement(By.xpath("./.."))
+            .findElement(By.xpath("./.."));
+
+    connectionItemParentParent
+            .findElement(By.cssSelector("div[class*='actions-popover']")).click();
```

Comment:
I think it's better to add more selectors to the html than to write a complex finding element logic

Reply:
yeah, added data-testid's wherever possible.